scl/apache/apache.conf

```diff
@@ -20,22 +20,53 @@
 #
 #############################################################################
 block parser apache-accesslog-parser(prefix(".apache.") template("${MESSAGE}")) {
+    channel {
+        rewrite {
+            set("`template`" value("1"));
+        };
 
+        # if log line begins with vhost:port
+        if {
+            filter { match("^[A-Za-z0-9\-\._]+:[0-9]+ " value("1")); };
+            parser {
+                csv-parser(
+                    dialect(escape-double-char)
+                    flags(strip-whitespace)
+                    delimiters(" ")
+                    template($1)
+                    quote-pairs('""[]')
+                    columns("2", "`prefix`clientip", "`prefix`ident",
+                            "`prefix`auth", "`prefix`timestamp",
+                            "`prefix`rawrequest", "`prefix`response",
+                            "`prefix`bytes", "`prefix`referrer",
+                            "`prefix`agent"));
 
-    channel {
-        parser {
-            csv-parser(
-                prefix(`prefix`)
-                dialect(escape-double-char)
-                flags(strip-whitespace)
-                delimiters(" ")
-                template(`template`)
-                quote-pairs('""[]')
-                # field names match of that of Logstash
-                columns("clientip", "ident", "auth",
-                        "timestamp", "rawrequest", "response",
-                        "bytes", "referrer", "agent"));
+                csv-parser(
+                    prefix(`prefix`)
+                    template("$2")
+                    delimiters(":")
+                    dialect(escape-none)
+                    columns("vhost", "port"));
+            };
 
+        # if traditional log format
```

Comment:
From clean code point of view, instead of this comment, you could give a name to these parser blocks. Analogously to
```
block parser apache-accesslog-parser(prefix(".apache.") template("${MESSAGE}"))
```
you could extract these branches into `apache-access-log-traditional-part` and `apache-access-log-combined-part`. That would make the scl a little more readable. As maybe there are other distributions with different format, it would be easier to extend later on.

Reply:
I wasn't aware that ubuntu has this LogFormat already available, thanks for pointing it out.  Fedora/CentOS does not include this format.  It might make more sense to reference the name as vhost_combined instead of vcombined.

Shifting from 1 -> 3 blocks is a good idea, I'll explore that tonight.